diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-core2.dts b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-core2.dts 
index c6fc096..731e2ae 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-core2.dts 
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-core2.dts
--- arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-core2.dts	2019-07-08 11:19:07.982615277 -0600
+++ arch/arm64/boot/dts/allwinner/sun50i-h5-nanopi-neo-core2.patched.dts	2019-07-08 16:01:07.963700875 -0600
@@ -45,6 +45,26 @@
 / {
 	model = "FriendlyElec NanoPi-NEO-Core2";
 	compatible = "friendlyelec,nanopi-neo-core2", "allwinner,sun50i-h5";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		pwr {
+			label = "nanopi:red:pwr";
+			gpios = <&r_pio 0 10 GPIO_ACTIVE_HIGH>; /* PL10 */
+			linux,default-trigger = "default-on";
+		};
+
+		status {
+			label = "nanopi:green:status";
+			gpios = <&pio 0 10 GPIO_ACTIVE_HIGH>; /* PA10 */
+			linux,default-trigger = "heartbeat";
+		};
+	};
 };
 
 &cpu0 {
@@ -159,10 +179,56 @@
 	vdd_cpu: regulator@65 {
 		compatible = "silergy,sy8106a";
 		reg = <0x65>;
+		silergy,fixed-microvolt = <1000000>;
 		regulator-min-microvolt = <940000>;
 		regulator-max-microvolt = <1400000>;
 		regulator-ramp-delay = <200>;
 		regulator-boot-on;
 		regulator-always-on;
 	};
-};
\ No newline at end of file
+};
+
+
+&ehci0 {
+	status = "okay";
+};
+
+&ehci3 {
+	status = "okay";
+};
+
+
+&ohci0 {
+	status = "okay";
+};
+
+&ohci3 {
+	status = "okay";
+};
+
+&r_i2c {
+	status = "okay";
+
+	vdd_cpux: regulator@65 {
+		compatible = "silergy,sy8106a";
+		reg = <0x65>;
+		silergy,fixed-microvolt = <1000000>;
+		regulator-min-microvolt = <1000000>;
+		regulator-max-microvolt = <1400000>;
+		regulator-ramp-delay = <200>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+};
+
+&uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_pins_a>;
+	status = "okay";
+};
+
+&usbphy {
+	/* USB Type-A ports' VBUS is always on */
+	status = "okay";
+};
+
